### DISCUSSION TITLE
full source code for Connect API section

### DISCUSSION THREAD

**Discussion by frankiee12a9** [reduxjs/react-redux#1862] [Q&A]
I'm new to Redux. Could you please provide the full source code for `Tutorial: Connect API` section?

**Comment by markerikson** [comment]
FWIW today we teach users to use the React-Redux hooks API by default.  `connect` still works, but we definitely don't want new users learning it unless they absolutely have to because of existing codebases.

Additionally, the tutorial for `connect` was written a few years ago, and it shows several usage patterns that are very outdated.  That tutorial actually needs to be replaced.

Please see the tutorials in our Redux core docs for the correct way to learn and use Redux Toolkit and React-Redux:

- https://redux.js.org/tutorials/index
